packages/@ember/routing/history-location.ts

```diff
@@ -1 +1,281 @@
-export { HistoryLocation as default } from '@ember/-internals/routing';
+import EmberObject, { set } from '@ember/object';
+import type { ILocation as EmberLocation, UpdateCallback } from '@ember/routing/location';
+import { getHash } from './lib/location-utils';
+
+/**
+@module @ember/routing
+*/
+
+let popstateFired = false;
+
+function _uuid() {
```

Comment:
Unrelated to this PR: we should seriously consider using [`Crypto.randomUUID()`](http://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID) for this… in a few years. Safari only just added support in 15.4. :weary:

Reply:
EmberData uses randomUUID with an optional polyfill (last 2 safari versions have randomUUID)

The bigger issue is that it is not available to insecure contexts (http)